tests/tests.js

```diff
@@ -1,14 +1,184 @@
+require('dotenv').config();
+
+let id;
+axios
+  .post(`${process.env.URI}/places`, { en: { name: 'guesthouse mario' } })
+  .then(res => {
+    id = res.data._id;
+  });
 
 tape('home route test: GET request to /', t => {
-  request(server)
+  supertest(server)
     .get('/')
     .end((err, res) => {
-      if (err) throw err;
+      t.error(err, '/ home route does not return an error');
       t.equal(res.status, 200, 'should return status code 200');
-      t.ok(res.text.includes('<div id="list-page-content">'), 'should return home page containing test string "<div id="list-page-content">"');
+      t.ok(res.text.includes('<div id="list-page-content">'));
+      t.end();
+    });
+});
+
+tape('Test POST arabic event', t => {
+  const data = {
+    name_arabic: 'شو عم بصير',
+    description_ar: 'واو',
+    categories: ['music'],
+    accessibilityOptions: ['Disabled toilets'],
+    startDate: '2017-11-24',
+    startTime: '00:00',
+    endTime: '14:00',
+    cost: 0,
+    place_name_ar: 'واو'
+  };
+
+  nock(process.env.URI)
+    .post('/events')
+    .reply(200, { _id: 1 });
+
+  nock(process.env.URI)
+    .post('/places')
+    .reply(200, { _id: 1 });
+
+  supertest(server)
+    .post('/add-event')
+    .send(data)
+    .end((err, res) => {
+      t.error(err, 'posting an event did not error');
+      console.log(res.statusCode);
+      t.end();
+    });
+});
+
+tape('Test POST english event', t => {
+  const data = {
+    name_english: 'going to the beach',
+    description_en: 'there is a beach',
+    categories: ['music'],
+    accessibilityOptions: ['Disabled toilets'],
+    startDate: '2017-11-24',
+    startTime: '00:00',
+    endTime: '14:00',
+    cost: 0,
+    place_name_en: 'over the rainbow'
+  };
+
+  nock(process.env.URI)
+    .post('/events')
+    .reply(200, { _id: 1 });
+
+  nock(process.env.URI)
+    .post('/places')
+    .reply(200, { _id: 1 });
+
+  supertest(server)
+    .post('/add-event')
+    .send(data)
+    .end((err, res) => {
+      t.error(err, 'posting an event did not error');
+      t.end();
+    });
+});
+
+tape('Test POST event with a specific place', t => {
+  const data = {
+    name_english: 'Finn week off',
+    description_en: 'there is a beach, music',
+    categories: ['music'],
+    accessibilityOptions: ['Disabled toilets'],
+    startDate: '2017-11-24',
+    startTime: '00:00',
+    endTime: '14:00',
+    cost: 0,
+    placeId: id
+  };
+
+  nock(process.env.URI)
+    .post('/events')
+    .reply(200, { _id: 1 });
+
+  nock(process.env.URI)
+    .post('/places')
+    .reply(200, { _id: null });
+
+  supertest(server)
+    .post('/add-event')
+    .send(data)
+    .end((err, res) => {
+      t.error(err, 'posting an event did not error');
+      t.end();
+    });
+});
+
+tape('Test the authentication middleware', t => {
+  nock(process.env.URI)
+    .get('/places')
+    .reply(200, [
+      { ar: { name: 'سيس' } },
+      { en: { name: 'base' } },
+      { en: { name: 'home' } }
```

Comment:
If you changed the map/sort in `get_places` to a helper function then you could get rid of this section and just test the function itself, rather than the route.

Reply:
still need to test the route that runs the function